vertx-web-api-service/pom.xml

```diff
@@ -113,10 +113,19 @@
             <goals>
               <goal>process</goal>
             </goals>
-            <phase>generate-sources</phase>
+            <phase>process-classes</phase>
             <configuration>
-              <!-- Don't run it on this project as it's a generator -->
-              <skip>true</skip>
+              <processors>
+                <processor>io.vertx.codegen.CodeGenProcessor</processor>
+                <processor>io.vertx.docgen.JavaDocGenProcessor</processor>
+                <processor>io.vertx.docgen.DocGenProcessor</processor>
```

Comment:
are you sure those aren't provided by the parent pom transitively ?

Reply:
They are provided but it's configured to run in `generate-sources`, while we must run it in `process-classes`. If you run it before compilation the codegen reads META-INF services and tries to load not compiled classes. I tried to exclude meta inf from cp without success. service-proxy uses a similar configuration: https://github.com/vert-x3/vertx-service-proxy/blob/master/pom.xml#L250

Reply:
i try it